Makefile.prow

```diff
@@ -0,0 +1,7 @@
+# Copyright Contributors to the Open Cluster Management project
+
+-include /opt/build-harness/Makefile.prow
+
+.PHONY: build
+build:
+	CGO_ENABLED=0 go build -o $(BINDIR)/search-aggregator ./
```

Comment:
```suggestion
	CGO_ENABLED=0 go build -o $(BINDIR)/search-aggregator ./
		
.PHONY: unit-test
unit-test:
	DEPLOYED_IN_HUB=true go test ./... -v -coverprofile cover.out
```

Reply:
Can we try to add unit-test to the makefile in this PR? It already looks good to merge. I don't want to make the PR too big, but it'd be nice to have this all in one spot

Reply:
Updated, also removed the `build` target because we don't use it from prow